src/main/java/me/coley/recaf/util/JavaParserUtil.java

```diff
@@ -506,7 +506,7 @@ private static String typeToDesc(Type type) {
 			ClassOrInterfaceType clsType = (ClassOrInterfaceType) type;
 			clsType.setTypeArguments(NO_TYPE_ARGS);
 			try {
-				key = toInternal(clsType.resolve().getTypeDeclaration().get());
+				key = toInternal(clsType.resolve().asTypeParameter());
```

Comment:
This will throw an exception when used:

```java
    default ResolvedTypeParameterDeclaration asTypeParameter() {
        throw new UnsupportedOperationException(String.format("%s is not a Type parameter", this));
    }
```

Did you mean to use `asReferenceType()`?